kong/conf_loader/listeners.lua

```diff
@@ -14,7 +14,7 @@ local listeners = {}
 
 
 local subsystem_flags = {
-  http = { "ssl", "http2", "proxy_protocol", "deferred", "bind", "reuseport",
```

Comment:
i'm thinking if we should do some auto convetsion on the `listen http2` pattern to `http2 on`, although this is not a breaking change and it's a nginx change, user would be confused when they see a warning pops up. how do you think?

Reply:
Yes, bumping to OpenResty 1.25.3.1 has many impacts to us, we should have more discussions.

Reply:
Done. Now we still can use the flag `http2` and it will be convert to directive `http2 on;` automaticly.